### PR TITLE
.github/pr-shell-check.yaml: add shell code scan CI

### LINKED ISSUE #6: add github action to scan shell (shellint)
https://github.com/intel/tdx-tools/blob/main/.github/workflows/pr-check-shell.yml

### PR DESCRIPTION
**Pull request by hwang37**
Create github action to check shell code, to fix issue #6

### CODE REVIEW

**Comment by leyao-daily** [comment]
Hi, @hwang37 Please create a new branch or based on the latest version, this PR contains the extra commit. And also create the commit and PR title with ".github/pr-xxxxxx: xxx`. Also keep the file name in same format with others. eg `pr-shell-check.yaml`

File: .github/workflows/pr-check-shell.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  workflow_dispatch:
+
+jobs:
+  codescan:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
+        with:
```
Comment (leyao-daily) [inline .github/workflows/pr-check-shell.yaml]: we don't have the path `build/ubuntu-22.04`, to modify the yaml to fit CNAP, pleaes refer to https://github.com/marketplace/actions/shellcheck

File: .github/workflows/pr-check-shell.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+          ignore_paths: build/ubuntu-22.04
```
Comment (leyao-daily) [inline .github/workflows/pr-check-shell.yaml]: add newline at the end of file.

File: .github/workflows/pr-check-shell.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: Check Shell Code
+
+on:
```
Comment (leyao-daily) [inline .github/workflows/pr-check-shell.yaml]: You can refer to other action yaml for this part, the push to main should trigger the action.

**Review by leyao-daily — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-check-shell.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+name: Check Shell Code
```
Comment (Hulongyin) [inline .github/workflows/pr-check-shell.yaml]: Keep the name in same format with other action yaml file for CNAP, also keep the file name in same format with other action yaml file.

File: .github/workflows/pr-check-shell.yaml (hunk 1 above)
Comment (hwang37) [reply .github/workflows/pr-check-shell.yaml]: Thanks, fixed.

File: .github/workflows/pr-check-shell.yaml (hunk 2 above)
Comment (hwang37) [reply .github/workflows/pr-check-shell.yaml]: Thanks, fixed.

File: .github/workflows/pr-check-shell.yaml (hunk 4 above)
Comment (hwang37) [reply .github/workflows/pr-check-shell.yaml]: Thanks, fixed

File: .github/workflows/pr-check-shell.yaml (hunk 3 above)
Comment (hwang37) [reply .github/workflows/pr-check-shell.yaml]: Thanks, fixed

**Comment by hwang37** [comment]
> Hi, @hwang37 Please create a new branch or based on the latest version, this PR contains the extra commit. And also create the commit and PR title with ".github/pr-xxxxxx: xxx`. Also keep the file name in same format with others. eg `pr-shell-check.yaml`

Thanks, fixed

**Review by Hulongyin — APPROVED** [review]
LGTM. Thanks

**Review by leyao-daily — APPROVED** [review]
LGTM. Thanks

**Pull request merged**